plugins/sip/lib/model/providers/pexip/pexipUtils.php

```diff
@@ -0,0 +1,310 @@
+<?php
+/**
+ * @package plugins.sip
+ * @subpackage model.pexip
+ */
+
+class PexipUtils
+{
+	/**
+	 * @return bool
+	 * @throws Exception
+	 * @throws kCoreException
+	 */
+	public static function initAndValidateConfig()
+	{
+		if (kConf::hasMap('sip') && $pexipConfig = kConf::get('pexip', 'sip'))
+		{
+			return $pexipConfig;
+		}
+
+		KalturaLog::debug("Pexip misconfigured. please validate configuration.");
+		return false;
+	}
+
+	/**
+	 * @param $dbLiveEntry
+	 * @param $pexipConfig
+	 * @param $regenerate
+	 * @return string
+	 */
+	public static function createSipUrl(LiveStreamEntry $dbLiveEntry, $pexipConfig, $regenerate = false)
+	{
+		if (!$dbLiveEntry->getSipToken() || $regenerate)
+		{
+			$addition = str_pad(substr((string)microtime(true)*10000, -5),5,'0',STR_PAD_LEFT);
+			$sipToken = $dbLiveEntry->getPartnerId() . $addition;
+			$dbLiveEntry->setSipToken($sipToken);
```

Comment:
This is a side effect.
The function should only create SipUrl and it must not update the state of the live entry